### DISCUSSION TITLE
Test an HDTV tuner card

### DISCUSSION THREAD

**Discussion by JanHBade** [geerlingguy/raspberry-pi-pcie-devices#23] [Ideas]
https://skinflint.co.uk/?cat=vid&xf=800_PCIe+x1

the cards from digital devices have a nice Linux support

**Comment by geerlingguy** [comment]
If I picked up one of these, I would probably want to first see if [tvheadend](https://github.com/tvheadend/tvheadend) will compile on ARM/Pi, and if not, it might not be worth trying to hack together the software side.

It seems like something like the [Hauppauge 1196 HVR-1265](https://amzn.to/397DX5A) would have the best shot at working, as it does have Linux drivers, though it's only 'supported' on Ubuntu. And I'd have to wonder if anyone's ever tested/compiled the drivers for ARM...

**Reply by SuNNjek** [reply]
There is an already compiled version of tvheadend available for arm64: https://www.raspberrypi.org/forums/viewtopic.php?f=63&t=275372&start=525#p1787800

**Reply by JanHBade** [reply]
tvheadend is not the problem, me running a docker container on a 64Bit Rasperry Pi OS, but the driver for the cards could be complicated
-> https://hub.docker.com/r/linuxserver/tvheadend

**Reply by 6by9** [reply]
TBS cards are annoying as they have no mainline drivers, so you'll be compiling out of tree modules. Building media-build means that it replaces all in-tree V4L2 drivers, so the flip side is that you then don't get any of the Pi downstream patches.
Looking at their drivers they appear to have used a custom FPGA or ASIC for their PCIe bridge, and that has no mainline support. It's not obvious how involved it would be to cherry-pick those across.

Hauppauge cards generally have reasonable support in mainline - one of their employees (Brad Love) was proactively upstreaming patches.

DVBSky cards likewise generally work, although mine seems to have died (see https://github.com/geerlingguy/raspberry-pi-pcie-devices/issues/57)

Yes, tvheadend just works on almost anything. It's packaged in Buster.
I have a 6 tuner (4 DVB-T2, 2 DVB-S2) USB rig set up here on a Pi4 that just chugs along happily.

**Comment by TheGuyDanish** [comment]
Chiming in since I've got a Pi Zero running it beautifully. Tvheadend works great with the Pi and the [TV hat](https://www.raspberrypi.org/products/raspberry-pi-tv-hat). The only challenge is that tvheadend's support has been a bit spotty with Raspbian Buster, and even Stretch is only for armhf. So using it with a newer OS and 64-bit could be a challenge.

Assuming there's a card with a neatly compatible driver, I could see this being a good use case.

~I'll go back to working on my mATX board now, though.~

**Comment by geerlingguy** [comment]
Hehe, can't wait to see a copy of that board :)

But extremely good to know—saves me the time I'd have to do testing it out as much as I can without a tuner card. I currently have a HDHomeRun so I might even test it out with that too.

**Comment by geerlingguy** [comment]
Anyone in here may also be interested in following the issue [Raspberry PI Working as a TV Reciver (TBS6522)](https://github.com/geerlingguy/raspberry-pi-pcie-devices/issues/57).

**Comment by KimHeist** [comment]
Hey All,
How does one unscramble tv channels?
